backend/src/modules/doubles-tournament/core/application/event/CallMatchWhenMatchWasQueued.ts

```diff
@@ -0,0 +1,36 @@
+import { EventHandler } from '../../../../../shared/core/application/event/EventHandler';
+import { MatchWasQueued } from '../../domain/event/MatchWasQueued';
+import { CommandPublisher } from '../../../../../shared/core/application/command/CommandBus';
+import { MatchesQueueRepository } from '../MatchesQueueRepository';
+import { TablesQueueRepository } from '../TablesQueueRepository';
+import { CallMatch } from '../command/CallMatch';
+
+export class CallMatchWhenMatchWasQueued implements EventHandler<MatchWasQueued> {
+  private readonly commandPublisher: CommandPublisher;
+  private readonly matchesQueueRepository: MatchesQueueRepository;
+  private readonly tablesQueueRepository: TablesQueueRepository;
+
+  constructor(
+    commandPublisher: CommandPublisher,
+    matchesQueueRepository: MatchesQueueRepository,
+    tablesQueueRepository: TablesQueueRepository,
+  ) {
+    this.commandPublisher = commandPublisher;
+    this.matchesQueueRepository = matchesQueueRepository;
+    this.tablesQueueRepository = tablesQueueRepository;
+  }
+
+  async handle(event: MatchWasQueued): Promise<void> {
+    const tournamentId = event.tournamentId;
+    const freeTablesQueue = await this.tablesQueueRepository.findFreeTablesByTournamentId(event.tournamentId);
+    if (freeTablesQueue && freeTablesQueue.length > 0) {
```

Comment:
Teraz się nie będziemy w to bawić, bo to trochę komplikuje rozwiązanie, ale czy widzisz tutaj jakieś zagrożenie?